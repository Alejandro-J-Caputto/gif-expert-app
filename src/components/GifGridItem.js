import React from 'react'
import PropTypes from 'prop-types';


export const GifGridItem = ({ id, title, url }) => {

    return (
        <div className="card animate__animated animate__fadeIn">
            <img src={url} alt={title} />
            <p key={id}>{title}</p>
        </div>
    )
}



GifGridItem.protoTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}

/*

Enzyme
Enzyme to JSON

 */

