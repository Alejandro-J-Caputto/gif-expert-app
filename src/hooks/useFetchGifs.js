import { useState, useEffect } from "react"
import { getGifs } from "../helpers/getGifs"



export const useFetchGifs = (categorie) => {

    const [state, setState] = useState({

        data: [],
        loading: true

    })

    useEffect(() => {
        getGifs(categorie).then(img =>

            setTimeout(() => {

                setState({
                    data: img,
                    loading: false
                });

            }, 3000))
    }, [categorie])



    return state; //data [], loading true.


}