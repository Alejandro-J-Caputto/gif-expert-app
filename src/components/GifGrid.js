
import React from 'react'
import PropTypes from 'prop-types';
import { GifGridItem } from './GifGridItem'
// import { getGifs } from '../helpers/getGifs'
import { useFetchGifs } from '../hooks/useFetchGifs'


export const GifGrid = ({ categorie }) => {



    // const [images, setImages] = useState([])

    // useEffect(() => {
    //     getGifs(categorie).then(gifs => {
    //         setImages(gifs)
    //     })
    // }, [category])


    //APLICANDO HOOK

    const { data: images, loading } = useFetchGifs(categorie)

    /////////////
    return (
        <>
            <h3 className="animate__animated animate__bounce">{categorie}</h3>
            {loading && <p>Loading</p>}

            {
                <div className="card-grid animate__animated animate__fadeIn">

                    {images.map(gif => {
                        return <GifGridItem
                            key={gif.id}
                            {...gif}

                        />

                    })}


                </div>
            }
        </>
    )
}
GifGrid.propTypes = {
    categorie: PropTypes.string.isRequired
}