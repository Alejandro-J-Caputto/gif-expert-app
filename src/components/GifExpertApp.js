

import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';


export const GifExpertApp = () => {

    const [categorias, setcategorias] = useState(['One Punch'])
    // const handleAdd = () => {
    //     // setcategorias([...categorias, 'Naruto'])
    //     setcategorias(cat =>
    //         [...categorias, 'Bleach']
    //     )
    // };


    return (
        <>
            <h1>GifExpertApp</h1>
            <AddCategory setcategorias={setcategorias} />
            <hr />

            {/* <button onClick={handleAdd} > agregar </button> */}
            <ol>
                {
                    categorias.map((categorie) => {
                        return <GifGrid key={categorie} categorie={categorie} />
                    })
                }
            </ol>
        </>
    )
}

export default GifExpertApp;