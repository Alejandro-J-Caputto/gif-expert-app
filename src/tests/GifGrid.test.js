import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from "../components/GifGrid";
import { useFetchGifs } from '../hooks/useFetchGifs';
jest.mock('../hooks/useFetchGifs')


describe('prueba en el GifGrid', () => {
    const categorie = 'One Punch'
    test('snapshot inicial', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow(<GifGrid categorie={categorie} />);

        expect(wrapper).toMatchSnapshot();

    })

    test('debe de mostrar items cuando se cargan imagenes con useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa',
            title: 'Cualquier Cosa'
        }, {
            id: 'ABCasda',
            url: 'https://localhost/cualquier/cosa',
            title: 'Cualquier Cosa'
        }
        ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });



        const wrapper = shallow(<GifGrid categorie={categorie} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });




})
