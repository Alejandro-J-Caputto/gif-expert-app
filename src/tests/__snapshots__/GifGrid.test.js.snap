// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba en el GifGrid debe de mostrar items cuando se cargan imagenes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce"
  >
    One Punch
  </h3>
  <div
    className="card-grid animate__animated animate__fadeIn"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier Cosa"
      url="https://localhost/cualquier/cosa"
    />
    <GifGridItem
      id="ABCasda"
      key="ABCasda"
      title="Cualquier Cosa"
      url="https://localhost/cualquier/cosa"
    />
  </div>
</Fragment>
`;

exports[`prueba en el GifGrid snapshot inicial 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce"
  >
    One Punch
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid animate__animated animate__fadeIn"
  />
</Fragment>
`;
